src/components/color/index.js

```diff
@@ -57,12 +61,23 @@ color.combine = function(front, back) {
     return tinycolor(fcflat).toRgbString();
 };
 
+/*
+ * Create a color that contrasts with cstr.
+ *
+ * If cstr is a dark color, we lighten it; if it's light, we darken.
+ *
+ * If lightAmount / darkAmount are used, we adjust by these percentages,
+ * otherwise we go all the way to white or black.
+ * TODO: black is what we've always done for hover, but should it be #444 instead?
+ */
 color.contrast = function(cstr, lightAmount, darkAmount) {
     var tc = tinycolor(cstr);
 
-    var newColor = tc.isLight() ?
-        tc.darken(darkAmount) :
-        tc.lighten(lightAmount);
+    if(tc.getAlpha() !== 1) tc = tinycolor(color.combine(cstr, '#fff'));
+
+    var newColor = tc.isDark() ?
+        (lightAmount ? tc.lighten(lightAmount) : '#fff') :
+        (darkAmount ? tc.darken(darkAmount) : '#000');
```

Comment:
I switched from `tc.isLight` to `tc.isDark` just because [`isLight` calls `isDark`](https://github.com/bgrins/TinyColor/blob/master/tinycolor.js#L56) anyway 🐎 .

Thoughts about my TODO?
> TODO: black is what we've always done for hover, but should it be #444 instead?

(ie `Color.defaultLine`)